### PR TITLE
Adding pointers to the web performance articles

### PR DESCRIPTION
**Pull request by sfshaza2**
Also doing some general cleanup.

John/Brett, I don't need both of you to review.  :)

### CODE REVIEW

File: src/docs/perf/rendering/best-practices.md [hunk 1]
```diff
@@ -24,7 +41,7 @@ when designing your app:
 * Avoid overly large single Widgets with a large `build()` function.
   Split them into different Widgets based on encapsulation but also on how
   they change:
-  * When `setState()` is called on a State, all descendent widgets will
```
Comment (lucalves) [inline src/docs/perf/rendering/best-practices.md:27]: I believe that in this case "will" is correct.

**Comment by sfshaza2** [comment]
@johnpryan?

**Comment by johnpryan** [comment]
Sorry, I didn't see this, LGTM

File: src/docs/perf/rendering/best-practices.md (hunk 1 above)
Comment (johnpryan) [reply src/docs/perf/rendering/best-practices.md:27]: I don't think it's adding any clarity here. The sentence is still correct without it.

**Comment by sfshaza2** [comment]
Thanks, @johnpryan!

**Pull request merged**